services/medium-editor-phrase.js

```diff
@@ -0,0 +1,217 @@
+    }
+    return selectionPhrases;
+  },
+
+  /**
+   * get the HTML from the selected range and either add or remove the phrases.
+   * @returns {string} HTML
+   */
+  togglePhrase: function () {
+    var html,
+      container = this.cloneSelection(),
+      selectionPhrases = this.getSelectionPhrases(container);
+
+    if (selectionPhrases.length) { // selection already has phrases, so remove them
+      selectionPhrases.forEach(this.removePhraseTags); // remove phrases while keeping their innerHTML
+      html = container.innerHTML;
+    } else { // no phrases found so add phrase tags
+      html = this.addPhraseTags(container.innerHTML);
+    }
+    return html;
+  },
+
+  /**
+   * replaces the selection with new html and selects the new html
+   * @param {string} html
+   */
+  replaceSelectionHtml: function (html) {
+    var range = MediumEditor.selection.getSelectionRange(this.document),
+      selection = this.document.getSelection(),
+      fragment;
+
+    range.deleteContents();
+    fragment = range.createContextualFragment(html);
+    range.insertNode(fragment);
+    if (fragment.firstChild) {
+      range.setStartBefore(fragment.firstChild);
+      range.setEndAfter(fragment.lastChild);
+    }
+    selection.removeAllRanges();
+    selection.addRange(range);
+  },
+
+  /**
+   * get the innerHTML or textContent
+   * @param {Node} node
+   * @returns {string}
+   */
+  getNodeHtml: function (node) {
+    switch (node.nodeType) {
+      case Node.ELEMENT_NODE:
+        return node.innerHTML;
+      case Node.TEXT_NODE:
+        return node.textContent;
+      default:
+        return node.innerHTML || node.textContent || '';
+    }
+  },
+
+  /**
+   * html before and after the selection remain phrases,
+   * a placeholder element becomes the selected range,
+   * and the selection html without phrase tags is returned.
+   * @param {Element} parentPhrase
+   * @returns {string}
+   */
+  removeParentPhrase: function (parentPhrase) {
+    var parentPhraseParent = parentPhrase.parentNode,
+      selectionHtml = this.getNodeHtml(this.cloneSelection()),
+      selection = this.document.getSelection(),
+      range = this.document.createRange(),
+      placeholderHtml = this.placeholderHtml;
+
+    this.replaceSelectionHtml(placeholderHtml); // replace the selection with a placeholder
+    parentPhrase.outerHTML = parentPhrase.cloneNode(true).innerHTML.split(placeholderHtml)
+      .map(phrase => phrase && this.addPhraseTags(phrase)) // add phrase tags before and after selection
+      .join(placeholderHtml); // re-insert placeholder where selection was
+
+    // select the placeholder
+    range.selectNode(parentPhraseParent.querySelector(this.placeholderSelector));
+    selection.removeAllRanges();
+    selection.addRange(range);
+
+    return selectionHtml;
+  },
+
+  /**
+   * traverse down from the selection to find at least one phrase
+   * @returns {boolean}
+   */
+  selectionContainsPhrase: function () {
```

Comment:
Might be easier to read if this function was named something like `hasSelectionPhrase` to convey a boolean return value.